tests/flask_integration/test_plan_repository.py

```diff
@@ -63,18 +63,30 @@ def test_that_availability_is_toggled_to_true(self) -> None:
         assert plan_from_repo
         assert plan_from_repo.is_available == True
 
-    def test_cannot_create_plan_from_non_existing_draft(self) -> None:
-        assert self.plan_repository.create_plan_from_draft(uuid4()) is None
-
-    def test_can_create_plan_from_exiting_draft(self) -> None:
-        draft = self.plan_generator.draft_plan()
-        assert self.plan_repository.create_plan_from_draft(draft.id) is not None
-
-    def test_query_plan_after_it_was_created_from_draft(self) -> None:
-        draft = self.plan_generator.draft_plan()
-        plan_id = self.plan_repository.create_plan_from_draft(draft.id)
-        assert plan_id
-        assert self.plan_repository.get_plans().with_id(plan_id)
+    def test_create_plan_propagates_specified_arguments_to_created_plan(self) -> None:
```

Comment:
I sort of disagree with checking that a row in the ``plan_review`` table was created but I agree with checking that the review process can go forward after creating a plan. That is why I created a test that checks that the approval date can be changed after the plan was created. I chose this approach since I think that the code will get inflexible if we check for the internals of the DB. Instead we should only test the interface of the implementation since this is what will be observable to the consumer of the DB interface, namely the business logic.

Unfortunately the DB interface is still very messy. Ideally we would want to create such a review object explicitly why the plan is reviewed and automatically when it is created. Imagine something like the following:

```
# entities.py
class PlanReview:
    plan_id: UUID
    is_approved: bool
    approval_date: datetime

# use_cases/approve_plan.py

```
class ApprovePlan:
    def __call__(self, request: Request) -> Response:
        ...
        self.database_gateway.create_plan_review(plan=plan.id, is_approved=True, approval_date=self.datetime_service.now())
        ...
```

Since I would like the application code to go into that direction I think that we should not test that the row in ``plan_review`` was created.